wasm/store.go

```diff
@@ -12,26 +12,67 @@ import (
 )
 
 type (
+	// Store is the runtime representation of "instantiated" Wasm module and objects.
+	// Multiple modules can be instantiated within a single store, and each instance,
+	// (e.g. function instance) can be referenced by other modules in a Store via imports.
+	//
+	// Every type whose name ends with "Instance" suffix belongs to exactly one store.
+	//
+	// Note that store is not thread (concurrency) safe, meaning that using single Store
+	// via multiple goroutines might result in race conditions. In that case, the invocation
+	// and access to any methods and field of Store must be guarded by mutex.
+	//
+	// See https://www.w3.org/TR/wasm-core-1/#store%E2%91%A0
 	Store struct {
-		engine          Engine
+		// The following fields are wazero-specific fields of Store.
+
+		// engine is a global context for a Store which is in reponsible for compilation and execution of Wasm modules.
+		engine Engine
+		// ModuleInstances holds the instantiated Wasm modules keyed on names given at Instantiate.
 		ModuleInstances map[string]*ModuleInstance
+		// TypeIDs maps each FunctionType.String() to a unique FunctionTypeID. This is used at runtime to
+		// do type-checks on indirect function calls.
+		TypeIDs map[string]FunctionTypeID
 
-		Functions []*FunctionInstance
-		Globals   []*GlobalInstance
-		Memories  []*MemoryInstance
-		Tables    []*TableInstance
-	}
+		// The followings fields match the definition of Store in the specification.
 
+		// Functions holds function instances (https://www.w3.org/TR/wasm-core-1/#function-instances%E2%91%A0),
+		// in this store.
+		// The slice index is to be interpreted as funcaddr(https://www.w3.org/TR/wasm-core-1/#syntax-funcaddr).
+		Functions []*FunctionInstance
+		// Globals holds global instances (https://www.w3.org/TR/wasm-core-1/#global-instances%E2%91%A0),
+		// in this store.
+		// The slice index is to be interpreted as globaladdr(https://www.w3.org/TR/wasm-core-1/#syntax-globaladdr).
+		Globals []*GlobalInstance
+		// Memories holds memory instances (https://www.w3.org/TR/wasm-core-1/#memory-instances%E2%91%A0),
+		// in this store.
+		// The slice index is to be interpreted as memoaddr(https://www.w3.org/TR/wasm-core-1/#syntax-memaddr).
+		Memories []*MemoryInstance
+		// Tables holds table instances (https://www.w3.org/TR/wasm-core-1/#table-instances%E2%91%A0),
+		// in this store.
+		// The slice index is to be interpreted as tableaddr(https://www.w3.org/TR/wasm-core-1/#syntax-tableaddr).
+		Tables []*TableInstance
+	}
+
+	// ModuleInstance represents instantiated wasm module.
+	// The difference from the spec is that in wazero, a ModuleInstance holds  pointers
+	// to the instances, rather than "addresses" (i.e. index to Store.Functions, Globals, etc) for convenience.
+	//
+	// See https://www.w3.org/TR/wasm-core-1/#syntax-moduleinst for the original specification.
```

Comment:
because you already clarified what isn't the spec and the spec link is intuitive

```suggestion
	// See https://www.w3.org/TR/wasm-core-1/#syntax-moduleinst
```